pkg/controllers/infracluster/infracluster_controller.go

```diff
@@ -356,3 +295,47 @@ func getReadiness(infraCluster client.Object) (bool, error) {
 
 	return val, nil
 }
+
+// getRawMAPIProviderSpec returns a raw Machine ProviderSpec from the the cluster.
+func getRawMAPIProviderSpec(ctx context.Context, cl client.Client) ([]byte, error) {
+	cpms, err := getActiveCPMS(ctx, cl)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get control plane machine set: %w", err)
+	}
+
+	if cpms == nil {
+		// The CPMS is not present or inactive.
+		// Devise VSphere providerSpec via one of the machines in the cluster.
+		machineSetList := &mapiv1beta1.MachineSetList{}
+		if err := cl.List(ctx, machineSetList, client.InNamespace(defaultMAPINamespace)); err != nil {
+			return nil, fmt.Errorf("%w: %w", errUnableToListMachineSets, err)
```

Comment:
I've never seen someone wrap two errors like this, I don't think this is what we want, I'd suggest replacing the first `%w` with just a string message